addons/OPT/REVIVE/fn_clientInitEH.sqf

```diff
@@ -115,21 +114,25 @@ DFUNC(playerHandleDamage) =
 	GVAR(playerHandleDamage_projectile) = _projectile; 
 	GVAR(playerHandleDamage_damage) = _damage; 
 
-	// Schade hoeher eingestellen Werte Blocken um Autorespwan zu verhindern.
-	if (_damage >= GVAR(MAX_DAMAGE)) then 
-    { 
-		_unit setDamage GVAR(MAX_DAMAGE);		
-	};	
+	private _resultingDamage = _damage;
 
-	if (getDammage _unit >= GVAR(MAX_DAMAGE)) then 
-    { 
-		_unit setDamage GVAR(MAX_DAMAGE);		
-	};	
+	if (_damage >= GVAR(MAX_DAMAGE)) then {   
+		[FUNC(playercheckINCAPACITATED), 1,""] call CLib_fnc_wait;
+		_resultingDamage = 0; 
+	};
 
-	[FUNC(playercheckINCAPACITATED), 1,""] call CLib_fnc_wait;
+	_resultingDamage;
 };
-
-player addEventHandler ["HandleDamage", FUNC(playerHandleDamage)];
+// omg dont look at me please
+player removeEventHandler ["HandleDamage", 0];
+player removeEventHandler ["HandleDamage", 1];
+player removeEventHandler ["HandleDamage", 2];
+player removeEventHandler ["HandleDamage", 3];
+player removeEventHandler ["HandleDamage", 4];
+player removeEventHandler ["HandleDamage", 5];
+player removeEventHandler ["HandleDamage", 6];
```

Comment:
Also wenn man das schon macht, dann sollte man dafür eine Schleife nehmen. Außerdem sollte ein Kommentar erklären warum das nötig ist.
Und bist du dir wirklich ganz sicher, dass das hier wirklich nötig ist?

Man könnte auch mal noch ausprobieren den EH nach dem Respawn verzögert hinzuzufügen (z.B. ne Sekunde oder so). Das sollte dann dafür sorgen dass euer EH immer der letzte EH ist, der hinzugefügt wird und somit sollte es dann derjenige sein, der den Schaden auch kontrollieren kann. Man kann den Spieler für diese Zeit ja auch unbesiegbar machen, um Nebeneffekte zu vermeiden (`player allowDamage false`).

```suggestion
for "_i" from 0 to 6 do {
    player removeEventHandler ["HandleDamage", _i];
};
```

Reply:
Foreach macht es kaputt. Dann wird der EH initial nicht hinzugefügt. Fahrscheinlich weil der ForI in nem anderen Scheduler läuft?